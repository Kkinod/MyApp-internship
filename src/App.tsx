import React from 'react'
import logo from './logo.svg'
import './App.css'
import ListWrapper from './components/ListWrapper/ListWrapper'
import Demo from './components/DemoMUI/Demo'


function App() {
	return (
		<div className='App'>
			 <Demo />
			<header className='App-header'>
				<h1>Meet ReqRes users!</h1>
				<h2>Application created with free ReqRes API</h2>
			</header>
			<ListWrapper />
		</div>
	)
}

export default App
