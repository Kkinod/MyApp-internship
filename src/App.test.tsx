import React from 'react'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import App from './App'

describe('Shapshot test', () => {
	it('shapshot for App component', () => {
		const renderedComponent = renderer
			.create(
				<MemoryRouter initialEntries={['/']}>
					<App />
				</MemoryRouter>
			)
			.toJSON()
		expect(renderedComponent).toMatchSnapshot()
	})
})

it('Testing "Page not found"', () => {
	const badRoute = '/bad/route'

	render(
		<MemoryRouter initialEntries={[badRoute]}>
			<App />
		</MemoryRouter>
	)

	expect(screen.getByText(/Page not found/i)).toBeInTheDocument()
})
