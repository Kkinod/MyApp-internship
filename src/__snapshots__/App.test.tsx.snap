// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapshot for App component 1`] = `
<div>
  <div
    className="MuiBox-root css-i9gxme"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-v526sl-MuiPaper-root-MuiAppBar-root"
    >
      <div
        className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular toolbar-container css-hyum1k-MuiToolbar-root"
      >
        <div
          className="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap typographyH5 css-vjc0tx-MuiTypography-root"
        >
          Meet ReqRes users!
          <div
            className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
          >
            Meet ReqRes users! Application created with free ReqRes API
          </div>
        </div>
      </div>
    </header>
  </div>
  <div
    className="main"
  >
    <div
      className="side-bar__container"
    >
      <div
        className="side-bar"
      >
        <div
          className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-10gsmdd-MuiTypography-root"
        >
          Give a smile to shiba!
        </div>
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-12kx30u-MuiAvatar-root"
          data-testid="avatar"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <button
          className="link"
        >
          Get new shiba image
        </button>
      </div>
    </div>
    <div
      className="wrapper"
    >
      <ul
        className="users-list"
      >
        <div
          className="users-list__container"
        >
          <div
            className="search css-1sxytl1"
            onChange={[Function]}
          >
            <div
              className="css-1iz4iln"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <div
              className="MuiInputBase-root MuiInputBase-colorPrimary css-7mt4h1-MuiInputBase-root"
              onClick={[Function]}
            >
              <input
                aria-label="search"
                className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Searchâ€¦"
                type="text"
              />
            </div>
          </div>
          <div
            className="users-list"
          >
            <button
              className="MuiLoadingButton-root MuiLoadingButton-loading MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root Mui-disabled css-1pkfx9i-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
              disabled={true}
              id=":r0:"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              type="button"
            >
              <span
                className="MuiLoadingButton-startIconLoadingStart MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SaveIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
                  />
                </svg>
              </span>
              <div
                className="MuiLoadingButton-loadingIndicator MuiLoadingButton-loadingIndicatorStart css-17xq7a1-MuiLoadingButton-loadingIndicator"
              >
                <span
                  aria-labelledby=":r0:"
                  className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
                  role="progressbar"
                  style={
                    Object {
                      "height": 16,
                      "width": 16,
                    }
                  }
                >
                  <svg
                    className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                    viewBox="22 22 44 44"
                  >
                    <circle
                      className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                      cx={44}
                      cy={44}
                      fill="none"
                      r={20.2}
                      strokeWidth={3.6}
                      style={Object {}}
                    />
                  </svg>
                </span>
              </div>
              User Not Found
            </button>
          </div>
        </div>
      </ul>
      <div
        className="abc"
      >
        <div
          aria-hidden="true"
          data-testid="custom-element"
        />
      </div>
    </div>
  </div>
</div>
`;
